### PR TITLE
Raise up PHP version

### PR DESCRIPTION
**Pull request by Gerych1984**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌

That PR is a part of [this](https://github.com/yiisoft/assets/pull/94)

1. Raise php version to ^8.0
2. Allow `RecursiveDirectoryIterator` as argument in `FileHelper::lastModifiedTime`

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Need a line for CHANGELOG.

**Comment by Gerych1984** [comment]
> We can merge both methods to one:
> 
> ```
> public static function lastModifiedTime(string|RecursiveDirectoryIterator ...$directories): ?int
> ```
> 
> Seems, this is will be convient method.

In current version that helper has php7.4 in composer. Without this i was use union types

**Comment by vjik** [comment]
> In current version that helper has php7.4 in composer. Without this i was use union types

Can raise PHP version to `^8.0` or use so:

```php
/**
 * @param string|RecursiveDirectoryIterator ...$directories
 */ 
public static function lastModifiedTime(...$directories): ?int
```

**Comment by Gerych1984** [comment]
> > In current version that helper has php7.4 in composer. Without this i was use union types
> 
> Can raise PHP version to `^8.0` or use so:
> 
> ```
> /**
>  * @param string|RecursiveDirectoryIterator ...$directories
>  */ 
> public static function lastModifiedTime(...$directories): ?int
> ```
 
I think raise version is a best solution. How do you think?

**Comment by vjik** [comment]
> I think raise version is a better solution, but it must be impemented in assets too. How do you think?

Yes, it's OK.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Yii Files Change Log
 
-## 1.1.0 under development
+## 1.1.1 under development
```
Comment (rustamwin) [inline CHANGELOG.md:3]:
```suggestion
## 2.0.0 under development
```

**Review by rustamwin — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Yii Files Change Log
 
-## 1.1.0 under development
+## 2.0.0 under development
+
+- Chg #64: Raise the minimum PHP version to 8.0 and allow `RecursiveDirectoryIterator` as argument in `FileHelper::lastModifiedTime` (@Gerych1984)
```
Comment (vjik) [inline CHANGELOG.md:5]:
```suggestion
- Chg #64: Raise the minimum PHP version to 8.0 and allow `RecursiveDirectoryIterator` as argument in `FileHelper::lastModifiedTime()` (@Gerych1984)
```

File: src/FileHelper.php [hunk 3]
```diff
@@ -513,40 +515,51 @@ private static function openDirectory(string $directory)
...
             $iterator = new RecursiveIteratorIterator(
-                new RecursiveDirectoryIterator($path, FilesystemIterator::SKIP_DOTS),
+                $path,
                 RecursiveIteratorIterator::SELF_FIRST
             );
 
-            foreach ($iterator as $p => $_info) {
-                $times[] = self::modifiedTime($p);
+            foreach ($iterator as $path => $_info) {
+                $timestamp = self::modifiedTime($path);
```
Comment (vjik) [inline src/FileHelper.php:554]: What happens if `RecursiveDirectoryIterator` is without option `SKIP_DOTS` ?

Maybe `modifiedTime()` must returns `int|null` also? `Null` for cases when filemtime returns `false`.

Comment (Gerych1984) [reply src/FileHelper.php:554]: > What happens if `RecursiveDirectoryIterator` is without option `SKIP_DOTS` ?
> 
> Maybe `modifiedTime()` must returns `int|null` also? `Null` for cases when filemtime returns `false`.

Interesting - it calculated filemtime from parent/root directory

**Comment by samdark** [comment]
👍

**Pull request merged**